### ISSUE TITLE
New clients connections are not randomized with multiple nodes

### ISSUE THREAD

**Issue by lbtrier** [nats-io/stan.java#145]
Hi!
We're having a small problem with the NATS Java Client. We have a cluster running 3 NATS nodes and all of our clients are Java apps.
We've noticed that even if we have 3 nodes, all client connections happen only in one node. So our first node has 600 connections and node 2 and 3 only 5 connections each. This is causing some load problems for us because we're dealing with almost 2 billion messages per month +-.
We're using Java client 2.2.3
Analysing the Java client code (`NatsConnection.java`), we noticed this (begins on line 183 of 2.2.3):

```java
void connect(boolean reconnectOnConnect) throws InterruptedException, IOException {

(...)

 for (String serverURI : getServers()) {
            if (isClosed()) {
                break;
            }
            this.connectError.set(""); // new on each attempt
           (...)
            tryToConnect(serverURI, System.nanoTime());
```
So basically, the nodes are not being randomized, and all connections are going to the first node on the `getServers()` array.
This behavior IS NOT seen on `reconnect()`, which randomises the connections, trying to balance a little
Is it possible to replicate the same behavior of `reconnect()` on `connect()`?

```java
Collection<String> buildReconnectList() {
        ArrayList<String> reconnectList = new ArrayList<>();

        reconnectList.addAll(getServers());

        if (options.isNoRandomize()) {
            return reconnectList;
        }

        Collections.shuffle(reconnectList);

        return reconnectList;
    }
```

**Comment by sasbury** [comment]
This is really a problem in nats.java, i will fix there.

**Comment by sasbury** [comment]
fixed in https://github.com/nats-io/nats.java/releases/tag/2.6.6
